Assignment6.Rmd

```diff
@@ -77,7 +83,26 @@ table(M2$certified, M2$predict2)
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
-
+library(readr)
+d1<-read_csv("student.record.csv")
+d2 <- d1[,c(3:11,13)]
+d3 <- na.omit(d2)
+#model 1
+d.tree1 <- rpart(as.factor(MAJOR1_DESCR) ~ .,  method="class", data=d3)
+printcp(d.tree1)
+post(d.tree1, file = "tree3.ps", title = "MOOC")
+p1<-predict(d.tree1, d3, type = "class")
+t1<-table(d3$MAJOR1_DESCR, p1)
+sum(diag(t1 ))/sum(t1 )
+#model 2
+d4 <- d3[,-1]
+d.tree2 <- rpart(SEX ~ .,  method="class", data=d4)
+printcp(d.tree2)
+post(d.tree2, file = "tree4.ps", title = "MOOC") 
+p2<-predict(d.tree2, d4, type = "class")
+t2<-table(d4$SEX, p2)
+sum(diag(t2))/sum(t2)
+#the second model is much better
```

Comment:
Nice work! It seems the second model improves a lot on precision of prediction, which would be great if you report in the findings. One thing you may want to pay attention to is that the two models are using different outcome variables, which could make it not very appropriate to compare directly here. When doing research with classification tree or other algorithms, you may want to use different feature/predictors to predict the same outcome and find the better model based on your research question, precision, parsimony and other measures/criteria of model comparison.